libsolidity/formal/SMTEncoder.cpp

```diff
@@ -867,6 +870,105 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const*>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const*>(_type)->valueType();
+		else
+			solAssert(false, "");
+	}
+	return dimensionCount;
+}
+
+bool isReturnedFromStruct(TypePointer _type)
+{
+	// So far it seems that only Mappings and ordinary Arrays are not returned (from the cases the compiles).
```

Comment:
The code in `ExpressionCompiler::appendStateVariableAccessor` confirms that

Reply:
I see it as well. I wonder if it can be refactored to a method that can be called both from `ExpressionCompiler` and from `SMTEncoder`.

Reply:
I think it's not worth extracting it, let's leave it as it is